plonky2/src/hash/merkle_tree.rs

```diff
@@ -37,8 +37,8 @@ impl<F: RichField, H: Hasher<F>> MerkleCap<F, H> {
         log2_strict(self.len())
     }
 
-    pub fn flatten(&self) -> Vec<F> {
-        self.0.iter().flat_map(|&h| h.to_vec()).collect()
+    pub fn flatten(&self) -> impl Iterator<Item = F> + '_ {
+        self.0.iter().flat_map(|h| h.into_iter())
```

Comment:
You can probably remove the closure and just hand over the function you want to call directly to `flat_map`.

Reply:
Unfortunately, I cannot. `h` is a reference and `into_iter` takes by value, which works in the closure because `h` is `Copy`. I could put a `copied` in here, but that seemed less readable.